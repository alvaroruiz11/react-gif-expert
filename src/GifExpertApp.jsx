import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  
  const [ categorias, setCategories ] = useState( ['One Punch'] );

  const onAddCategory = ( newCategory ) => {

    if( categorias.includes( newCategory ) ) return;

    setCategories([
      newCategory,
      ...categorias
    ]);
  }

  return (
    <>
        <h1>GifExpertApp</h1>

        <AddCategory 
          onNewCategory={ ( value ) => onAddCategory( value ) }
        />
        <ol>
          {
            categorias.map( c => (
              <GifGrid 
                key={ c } 
                category={ c }
              />
            ))
          }
        </ol>

    </>
  )
}
