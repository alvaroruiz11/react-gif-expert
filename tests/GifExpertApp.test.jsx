import { render, screen } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Prueba en <GifExpertApp/>', () => {

    test('debe hacer macth con el snapshot', () => {

        const { container } = render( <GifExpertApp/> );

        expect( container ).toMatchSnapshot();
    });

    test('debe de haber el titulo de la aplicacion', () => {
        
        render( <GifExpertApp/> );

        expect( screen.getByText('GifExpertApp')).toBeTruthy();
    });

});