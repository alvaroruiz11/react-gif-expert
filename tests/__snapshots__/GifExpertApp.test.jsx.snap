// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifExpertApp/> debe hacer macth con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <ol>
    <h3>
      One Punch
    </h3>
    <h2>
      Cargando...
    </h2>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
